sycl/include/sycl/ext/oneapi/backend/level_zero.hpp

```diff
@@ -141,10 +74,11 @@ inline queue make_queue<backend::ext_oneapi_level_zero>(
                                 : reinterpret_cast<pi_native_handle>(
                                       *(std::get_if<ze_command_queue_handle_t>(
                                           &BackendObject.NativeHandle)));
-  return ext::oneapi::level_zero::make_queue(
-      TargetContext, Device, Handle, IsImmCmdList,
+
+  return sycl::detail::make_queue(
+      Handle, IsImmCmdList, TargetContext, &Device,
       BackendObject.Ownership == ext::oneapi::level_zero::ownership::keep,
-      BackendObject.Properties);
+      BackendObject.Properties, Handler, backend::ext_oneapi_level_zero);
```

Comment:
Before this change, we were using `async_handler` from the `TargetContext`, see https://github.com/intel/llvm/blob/3ea29b2a9028b485b76339e16754e3e74c9cc7a6/sycl/source/backend/level_zero.cpp#L70-L72. I'm updating to use the provided `Handler`, although I'm not sure about the consequences of that change. Regardless, using provided `Handler` instead of doing something completely different looks like a right thing to do for this interface.